magento/pwa-studio
# [PB-PWA-2064]: Add Google reCAPTCHA V3 validation to customer account forms

## Description

PR to add Google reCaptchaV3 validation to the following forms in Venia storefront:

- Customer login
- Create new customer
- Edit customer account

## Related Issue







Closes PWA-2064

## Acceptance




A normal user (non robot) should be able to submit all form requests and pass reCaptcha validation indicated by a successful request from graphQL. An automated user (robot), should not pass validation and graphQL should return an error.

### Verification Stakeholders
@schensley @tkacheva 


### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature





1. Verify that Google reCAPTCHA V3 is enabled in Admin for the following forms:
- Customer login
- Create new customer account
- Edit customer account

2. Access venia store
3. Submit create new customer form
4. Verify graphQL response indicates success
5. Open Account menu and edit customer info
6. Submit edit customer form
7.  Verify graphQL response indicates success
8. Logout and sign in with same customer
9. Verify graphQL response indicates success

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations


10. In Admin, set reCAPTCHA V3 minimum score threshold to 1.0
11. Open chrome dev tools, click on "More tools", select network conditions, change User agent to Googlebot.
12. Repeat same steps above, all submited forms should fail reCaptcha validation and return errors.



#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
